examples/gameroom/gameroom-app/src/views/GameroomDetail.vue

```diff
@@ -95,65 +95,57 @@ limitations under the License.
             :game="game" />
          </div>
          <div class="placeholder-wrapper" v-else>
-           <h3 class="tbl-placeholder"> {{ placeholderText }} </h3>
-           <div v-if="gameroom.status !== 'Active' || loadingGames" class="spinner-gameroom spinner" />
+           <h3 class="tbl-placeholder">No games to show</h3>
          </div>
-       </div>
+      </div>
+    <div class="data-container" v-else>
+      <loading message="Please wait while the gameroom is being set up" />
+    </div>
   </div>
 </template>
 
 <script lang="ts">
 import { Vue, Component } from 'vue-property-decorator';
 import gamerooms from '@/store/modules/gamerooms';
-import selectedGameroom from '@/store/modules/selectedGameroom';
-import games from '@/store/modules/games';
 import { gameIsOver, userIsInGame, userCanJoinGame} from '@/utils/xo-games';
 import { Gameroom, Member, Game } from '@/store/models';
 import Modal from '@/components/Modal.vue';
 import GameCard from '@/components/GameCard.vue';
 import store from '@/store';
+import Loading from '@/components/Loading.vue';
 
 @Component({
-  components: { Modal, GameCard },
+  components: { Modal, GameCard, Loading },
 })
   export default class GameroomDetails extends Vue {
       gameNameFilter = '';
       currentTab = 1;
       newGameName = '';
       displayModal = false;
       submitting = false;
-      loadingGames = false;
-
-      beforeRouteEnter(to: any, from: any , next: any) {
-        setPageLoading(true);
 
-        store.dispatch('selectedGameroom/updateSelectedGameroom', to.params.id).catch((e) => {
-            setPageLoading(false);
-            next({ name: 'not-found' });
-        }).then(() => {
-            setPageLoading(false);
-            next((vm: any) => vm.listGames(to.params.id));
-        });
-      }
-
-      beforeRouteUpdate(to: any, from: any , next: any) {
-        setPageLoading(true);
-        store.commit('pageLoading/setPageLoading', true);
-        store.dispatch('selectedGameroom/updateSelectedGameroom', to.params.id).catch((e) => {
-            setPageLoading(false);
-            next({ name: 'not-found' });
-        }).then(() => {
-            setPageLoading(false);
-            this.listGames(to.params.id);
-            next();
-        });
+      async beforeRouteEnter(to: any, from: any , next: any) {
```

Comment:
Could a type be created for a route to avoid using the any type here?

Reply:
Probably should be, but that would be a separate improvement since the only change to this line is making the function async

Reply:
After talking in person, decided to do this in a later change